pi/utils/dynamo.py

```diff
@@ -0,0 +1,12 @@
+import pyframe_eval
+
+
+class dynamo:
+    def __init__(self, callback):
+        self.callback = callback
+
+    def __enter__(self):
+        pyframe_eval.set_eval_frame(None, self.callback)
```

Comment:
One thing I don't understand is the use of this first parameter, as even in pyframe_eval it doesn't seem to be used:
https://github.com/makslevental/pyframe_eval/blob/df4f73dcf911e6f7654336f07351482c5eb08e1a/evalFrame.cpp#L254

Or is this a placeholder for something in the future?

Reply:
I don't really understand this either but for some reason there's an implicit `self` for the module - notice `dummy` everywhere in [torch/csrc/dynamo/eval_frame.c](https://github.com/pytorch/pytorch/blob/b5c24041166581ff0f38198d5ade63983c1e536d/torch/csrc/dynamo/eval_frame.c#L755-L784).